mdx-web/src/main/java/com/mx/path/model/mdx/web/controller/AuthenticationController.java

```diff
@@ -29,8 +32,66 @@
 @RequestMapping(value = "{clientId}", produces = BaseController.MDX_MEDIA)
 public class AuthenticationController extends BaseController {
 
-  @RequestMapping(value = { "/authentications", "/sessions" }, method = RequestMethod.POST, consumes = { MDX_MEDIA, MDX_ONDEMAND_MEDIA }, produces = { MDX_MEDIA, MDX_ONDEMAND_MEDIA })
-  public final ResponseEntity<Authentication> authenticate(@PathVariable("clientId") String clientId, @RequestBody Authentication requestAuthentication) {
+  @SuppressWarnings("MagicNumber")
+  @RequestMapping(value = "/authentications", method = RequestMethod.POST, consumes = MDX_MEDIA)
+  public final ResponseEntity<?> authenticate(@PathVariable("clientId") String clientId, HttpServletRequest request) {
+    return versioned(request)
+        .defaultVersion(Authentication.class, Authentication.class, (authentication -> {
+          return authenticateLegacy(clientId, authentication);
+        }))
+        .version(20240209, Authentication20240209.class, Authentication20240209.class, (authentication -> {
+          return authenticate(clientId, authentication);
+        }))
+        .execute();
+  }
+
+  @RequestMapping(value = { "/sessions" }, method = RequestMethod.POST, consumes = MDX_ONDEMAND_MEDIA, produces = MDX_ONDEMAND_MEDIA)
+  public final ResponseEntity<Authentication20240209> authenticate(@PathVariable("clientId") String clientId, @RequestBody Authentication20240209 requestAuthentication) {
```

Comment:
It would be nice to reduce code duplication where possible. I can see this getting out of hand if multiple model versions need to be supported in various methods.

Reply:
Agreed. However, the authentication controller is unique. Most controllers just invoke the gateway method and don't have this much code (which is the way we want it). I don't think this will pose an issue if used in most other controllers.